### PR TITLE
Milestone: Add forms

### PR DESCRIPTION
**Pull request by NickEmma**
# Hi there 👋🏻

**Thanks for taking the time to review the project. Here are some things important to note about it:**

## Changes implemented in the "App forms" branch

- ✅ Add a form to create a new post on behalf of the current_user. This form exists on another path posts/new
- ✅ Add a form to create a new comment on behalf of the current_user. This form exists within each post
- ✅ Add a form to generate a like button. This form exists in both the user_posts path and the user_post path
- ✅ Add validation to only allow one like button

# Final notes 📔

**Is expected that this project meets the requirements. Please let me know if anything is left to be done or can be improved. Again, thanks for the review.**

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      flash[:error] = 'Comment not created'
+    end
+    redirect_to user_posts_path(current_user)
+  end
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:text)
+  end
```
Comment (DeliceLydia) [inline app/controllers/comments_controller.rb:18]: - [ ] I am not able to see the comments form please create a `new` method 
```ruby
 def new
    @comment = Comment.new
  end
```

- [ ] and create a route something like:
```ruby
  POST   /posts/:post_id/comments(.:format)     comments#create
 GET   /posts/:post_id/comments/new(.:format)     comments#new
```

File: app/controllers/likes_controller.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  def create
+    post = Post.find(params[:post_id])
+    like = current_user.likes.new(post:)
+    if like.save!
+      flash[:success] = 'Post Liked successfully'
+    else
+      flash[:error] = 'Like not added'
+    end
+    redirect_to user_posts_path(current_user)
+  end
```
Comment (DeliceLydia) [inline app/controllers/likes_controller.rb:11]: - [ ] I am not able to like a post please fix that I think the issue is where you are creating a like I think it should be:
`like = current_user.likes.new(post_id: post.id)`

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @NickEmma, 

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

File: app/controllers/likes_controller.rb (hunk 2 above)
Comment (NickEmma) [reply app/controllers/likes_controller.rb:11]: noted thanks

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (NickEmma) [reply app/controllers/comments_controller.rb:18]: thank you for your review. i will kindly fix that

**Review by DammyShittu — APPROVED** [review]
Hi @NickEmma,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**